### PR TITLE
Break celery worker and beat into their own pod

### PR DESCRIPTION
**Pull request by WillNilges**
Moves celery out of the main meshdb pod and into its own. The beat and worker each get their own container within that pod, and everyone communicates through Redis.

Inspired by: https://medium.com/ambient-innovation/health-checks-for-celery-in-kubernetes-cf3274a3e106

### CODE REVIEW

**Comment by WillNilges** [comment]
https://medium.com/ambient-innovation/health-checks-for-celery-in-kubernetes-cf3274a3e106

**Review by Andrew-Dickinson — CHANGES_REQUESTED** [review]
Can we move these celery related scripts into sub-directories? It's a lot of clutter for the root of the repo

**Comment by Andrew-Dickinson** [comment]
Happy to override codecov here. Testing celery statically is hard. With Datadog will we be able to monitor to ensure the tasks are running and alert if they stop?

File: .github/workflows/publish-and-deploy.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Publish and Deploy
 
 on:
   push:
-    branches: [ main ]
```
Comment (Andrew-Dickinson) [inline .github/workflows/publish-and-deploy.yaml:6]: Did you mean to leave this changed?

File: .github/workflows/publish-and-deploy.yaml [hunk 2]
```diff
@@ -69,7 +69,7 @@ jobs:
       environment: dev3
     secrets: inherit
     needs: push_to_registry
-    if: github.ref == 'refs/heads/main'
```
Comment (Andrew-Dickinson) [inline .github/workflows/publish-and-deploy.yaml:72]: This too?

File: entrypoint.sh [hunk 3]
```diff
@@ -11,19 +11,6 @@ done
 
 echo 'DB started'
 
-# It's okay to start Celery in the background and continue without waiting, even though "migrate"
-# might make DB changes we want to notify for since tasks are queued by Django Webhook and
-# are executed as soon as celery starts
-# FIXME: This makes testing locally a bit awkward, since this isn't started by "manage.py runserver"
-#  maybe there's a way to do this better?
```
Comment (Andrew-Dickinson) [inline entrypoint.sh:18]: How does this comment change with these changes? Is the guidance still to run these commands from the readme for local dev?
```
celery -A meshdb worker -l INFO
celery -A meshdb beat -s /tmp/celerybeat-schedule -l DEBUG
```

Also does the celery worker we spin up programmatically during unit testing need to be changed in any way?

Comment (WillNilges) [reply entrypoint.sh:18]: Shouldn't be. This just moves the worker and beat out of the prod container. Grandbox will change, but whatever. I guess what I'd do if I were testing this locally would be to run the provided scripts.

**Comment by WillNilges** [comment]
> Can we move these celery related scripts into sub-directories? It's a lot of clutter for the root of the repo
> 
> Perhaps:
> 
> * bin/
>   
>   * celery/
>     
>     * All the celery stuff
>   * All the other scripts we're collecting

@Andrew-Dickinson I agree. I was going to make a `scripts/` directory in a follow-up PR, but I'm happy to do it in this one.

File: .github/workflows/publish-and-deploy.yaml (hunk 2 above)
Comment (WillNilges) [reply .github/workflows/publish-and-deploy.yaml:72]: Gosh you're quick on the draw

Comment (Andrew-Dickinson) [reply .github/workflows/publish-and-deploy.yaml:72]: Fastest click in the west 🤠

**Comment by WillNilges** [comment]
> Happy to override codecov here. Testing celery statically is hard. With Datadog will we be able to monitor to ensure the tasks are running and alert if they stop? (and can we document that requirement, in #88 perhaps)

Certainly. Feel free to add that to the reqs.

File: entrypoint.sh (hunk 3 above)
Comment (Andrew-Dickinson) [reply entrypoint.sh:18]: Could you update the readme with the latest guidance?

Comment (WillNilges) [reply entrypoint.sh:18]: Fuck, I knew you were gonna say that.

Comment (WillNilges) [reply entrypoint.sh:18]: Actually, on second thought, the readme instructions are fine. What's there right now will still work.

File: src/meshdb/celery.py [hunk 4]
```diff
@@ -13,37 +13,37 @@
 class LivenessProbe(bootsteps.StartStopStep):
     requires = {"celery.worker.components:Timer"}
 
-    def __init__(self, worker, **kwargs):
+    def __init__(self, worker, **kwargs):  # type: ignore[no-untyped-def]
```
Comment (Andrew-Dickinson) [inline src/meshdb/celery.py:16]: We can probably do types here, just use `dict` for the `kwargs` and `None` for the return type

**Review by Andrew-Dickinson — APPROVED** [review]
Approved, but I do think you can do better than that on the MyPy types and I would like to see _some_ improvement over what we have now.

If you're really going to ignore that many lines you should probably just exclude the whole file

File: src/meshdb/celery.py (hunk 4 above)
Comment (WillNilges) [reply src/meshdb/celery.py:16]: Didn't know that, thanks.

Comment (Andrew-Dickinson) [reply src/meshdb/celery.py:16]: Worker may be tricky but if it has a class you can just use the class and it’s essy

**Comment by WillNilges** [comment]
_maybe_ we can get the redefinitions to stop if we break out the checks into other files, like in the blog post I attached in the description.

**Pull request merged**